lib/rollout.rb

```diff
@@ -1,28 +1,30 @@
 require "rollout/version"
 require "zlib"
 require "set"
+require "json"
 
 class Rollout
   class Feature
-    attr_accessor :groups, :users, :percentage
+    attr_accessor :groups, :users, :percentage, :data
     attr_reader :name, :options
 
     def initialize(name, string = nil, opts = {})
       @options = opts
       @name    = name
 
       if string
-        raw_percentage,raw_users,raw_groups = string.split("|")
+        raw_percentage,raw_users,raw_groups,raw_data = string.split(/(?<!\\)\|/)
         @percentage = raw_percentage.to_f
         @users = (raw_users || "").split(",").map(&:to_s).to_set
         @groups = (raw_groups || "").split(",").map(&:to_sym).to_set
+        @data = raw_data ? JSON.parse(raw_data) : {}
```

Comment:
this should be `@data = raw_data.blank? ? {} : JSON.parse(raw_data)` if `String#blank?` exists to avoid

```
[14] pry(main)> JSON.parse('')
JSON::ParserError: A JSON text must at least contain two octets!
from /Users/rene/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/json_pure-1.8.3/lib/json/common.rb:155:in `initialize'
```